### PR TITLE
tetrio-desktop: re-add tetrio-plus, now from source

### PR DESCRIPTION
**Pull request by huantianad**
## Description of changes
Depends on #300166

The Tetrio plus support was removed with the update to v9, this adds it backing based off of Tetrio plus nightly with has v9 support, and we now build from source too.

@wackbyte 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix
Comment (huantianad) [inline pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix:98]: this phase seperation is really abitrary, maybe it can be combined.

File: pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix
Comment (huantianad) [inline pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix:106]: some jank permissions stuff with how im installing files, same with the install command above

File: pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix
Comment (wackbyte) [inline pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix:20]: Most packages seem to write this as `unstable-YYYY-MM-DD` with the date of the revision used?

File: pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix
Comment (wackbyte) [reply pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix:106]: I think someone with more experience than me should review this part.

**Review by wackbyte — CHANGES_REQUESTED** [review]

**Comment by wackbyte** [comment]
It looks like there's some trailing whitespace causing the EditorConfig check to fail.

File: pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix
Comment (wackbyte) [inline pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix:29]:
```suggestion
    # tetrio-plus uses this info for displaying its version,
```

File: pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix
Comment (wackbyte) [inline pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix:122]:
```suggestion
    # This patches the vanilla code to load the tetrio-plus code
```

File: pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix
Comment (wackbyte) [inline pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix:148]:
```suggestion
    description = "TETR.IO customization tool suite";
```

**Review by wackbyte — CHANGES_REQUESTED** [review]

**Comment by huantianad** [comment]
@wackbyte should I mark you as maintainer for this as well?

File: pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix
Comment (huantianad) [inline pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix:124]: Maybe we should reimplement this script on our side, since it does some things for say uninstalling that we can't do in the nix package anyway.

**Comment by wackbyte** [comment]
> @wackbyte should I mark you as maintainer for this as well?

Sure!

**Comment by wackbyte** [comment]
I'm not sure if there's a way to do that, unfortunately

**Comment by huantianad** [comment]
I mentioned to the maintainer of tetrio-plus about having them include the modified package.json as well as a lock file on their side, so might be worth waiting for that before including this.

**Pull request merged**
